src/Api/Controllers/OrganizationUsersController.cs

```diff
@@ -270,8 +270,17 @@ public async Task PutGroups(string orgId, string id, [FromBody] OrganizationUser
         [HttpPut("{userId}/reset-password-enrollment")]
         public async Task PutResetPasswordEnrollment(string orgId, string userId, [FromBody] OrganizationUserResetPasswordEnrollmentRequestModel model)
         {
+            var user = await _userService.GetUserByIdAsync(new Guid(userId));
```

Comment:
I think userId and  the `User` principal are the same here (as verified later in `UpdateUserResetPasswordEnrollmentAsync`. In that case, you could optimize this a bit more and just do

```
            var callingUser = await _userService.GetUserByPrincipalAsync(User);
```

and get rid of `_userService.GetProperUserId(User);` below

Reply:
Got it - modified to:
`var callingUser = await _userService.GetUserByPrincipalAsync(this.User);
            await _organizationService.UpdateUserResetPasswordEnrollmentAsync(
                new Guid(orgId), new Guid(userId), model.ResetPasswordKey, callingUser.Id);`